spec/lib/aganakti/client_spec.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+RSpec.describe Aganakti::Client do
+  describe '.new' do
+    it 'freezes the options' do
+      options = {}
+
+      expect { described_class.new('http://localhost', options) }.to change { options.frozen? }.from(false).to(true)
+    end
+
+    it 'freezes the URI' do
+      uri = String.new('http://localhost')
+
+      expect { described_class.new(uri, {}) }.to change { uri.frozen? }.from(false).to(true)
+    end
+
+    it 'overrides Accept header' do
+      client = described_class.new('http://localhost', headers: { 'Accept' => 'text/xml' })
+
+      expect(client.typhoeus_options[:headers]).to include('Accept' => 'application/json')
+    end
+
+    it 'overrides Content-Type header' do
+      client = described_class.new('http://localhost', headers: { 'Content-Type' => 'text/xml' })
+
+      expect(client.typhoeus_options[:headers]).to include('Content-Type' => 'application/json')
+    end
```

Comment:
is it worth making sure any other header passed survives? eg. `headers: { 'foo' => 'bar' }` will still be there next to Accept and Content-Type.